Detect k3s cluster and install portworx accordingly

piyush-nimbalkar opened the pull request:
- add extra mount for k3s
- enable csi

Signed-off-by: Piyush Nimbalkar <piyush@portworx.com>

zoxpx commented on drivers/storage/portworx/deployment.go:
```diff
@@ -878,6 +895,10 @@ func (t *template) getEnvList() []v1.EnvVar {
+		volumeInfoList = append(volumeInfoList, t.getK3sVolumeInfoList()...)
+	}
+
 	volumeMounts := make([]v1.VolumeMount, 0)
```
nit -- the following allocation is a little bit better:
```go
volumeMounts := make([]v1.VolumeMount, 0, len(volumeInfoList))
```
* I normally always take advantage of the fact that we know exactly how large will the produced array be
* if we leave it up to `append()` to extend the array-capacity, then it'll _double_ the array capacity (also _copy_ all the elements) every time it extends the capacity

piyush-nimbalkar commented:
Addressed both the comments.
Introduced an internal annotation on StorageCluster `portworx.io/run-on-master` which users can use to toggle running on master nodes. With k3s we will run on master by default without user adding the annotation.
